firewood/src/db.rs

```diff
@@ -381,6 +381,15 @@ impl<S: ShaleStore<Node> + Send + Sync> DbRev<S> {
     }
 }
 
+impl DbRev<MutStore> {
+    pub fn into_shared(self) -> DbRev<SharedStore> {
+        DbRev {
+            header: self.header,
+            merkle: self.merkle.into(),
+        }
+    }
+}
```

Comment:
If you just implement From, you get into() for free. This change plus changing proposal.rs to call into() instead of into_shared() just works.
```suggestion
impl From<DbRev<MutStore>> for DbRev<SharedStore> {
    fn from(value: DbRev<MutStore>) -> Self {
        DbRev {
            header: value.header,
            merkle: value.merkle.into(),
        }
    }
}
```